std/complex.d

```diff
@@ -973,6 +973,187 @@ Complex!T tan(T)(Complex!T z) @safe pure nothrow @nogc
     assert(ceqrel(tan(complex(0.0, 1.0)), complex(0.0, std.math.tanh(1.0))) >= double.mant_dig - 2);
 }
 
+/**
+    Inverse trigonometric functions on complex numbers.
+
+    Params: z = A complex number.
+    Returns: The arcsine, arccosine and arctangent of `z`, respectively.
+*/
+Complex!T asin(T)(Complex!T z)  @safe pure nothrow @nogc
+{
+    auto ash = asinh(Complex!T(-z.im, z.re));
+    return Complex!T(ash.im, -ash.re);
+}
+
+///
+@safe pure nothrow unittest
+{
+    import std.math : isClose, PI;
+    assert(asin(complex(0.0)) == 0.0);
+    assert(isClose(asin(complex(0.5L, 0)), PI / 6));
+    assert(isClose(asin(complex(0.5f, 0)), float(PI) / 6));
+}
+
+/// ditto
+Complex!T acos(T)(Complex!T z)  @safe pure nothrow @nogc
+{
+    static import std.math;
+    auto as = asin(z);
+    return Complex!T(std.math.PI_2 - as.re, as.im);
+}
+
+///
+@safe pure nothrow unittest
+{
+    import std.math : isClose, PI, std_math_acos = acos;
+    assert(acos(complex(0.0)) == std_math_acos(0.0));
+    assert(isClose(acos(complex(0.5L)), PI / 3));
+    assert(isClose(acos(complex(0.5f)), float(PI) / 3));
+}
+
+/// ditto
+Complex!T atan(T)(Complex!T z) @safe pure nothrow @nogc
+{
+    static import std.math;
+    const T re2 = z.re * z.re;
+    const T x = 1.0 - re2 - z.im * z.im;
```

Comment:
See https://d.godbolt.org/z/Efrqf6 as to why `1.0` is still bad here.

Reply:
Using integers where possible it is then.